### PR TITLE
Polish the Azure.ResourceManager.Maintenance

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs [hunk 1]
```diff
@@ -112,108 +75,117 @@ public partial class MaintenanceConfigurationResource : Azure.ResourceManager.Ar
...
+        public static Azure.Response<Azure.ResourceManager.Maintenance.MaintenanceApplyUpdateResource> CreateOrUpdateApplyUpdateByParent(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource, string providerName, string resourceParentType, string resourceParentName, string resourceType, string resourceName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Maintenance.MaintenanceApplyUpdateResource>> CreateOrUpdateApplyUpdateByParentAsync(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource, string providerName, string resourceParentType, string resourceParentName, string resourceType, string resourceName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs:81]: Yes, the swagger is weird which is not RESTFul, so I'd like to leave it as is

File: sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs [hunk 2]
```diff
@@ -112,108 +75,117 @@ public partial class MaintenanceConfigurationResource : Azure.ResourceManager.Ar
...
-        public virtual Azure.Response<Azure.ResourceManager.Maintenance.PublicMaintenanceConfigurationResource> Get(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Maintenance.PublicMaintenanceConfigurationResource>> GetAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.Maintenance.MaintenancePublicConfigurationResource> Get(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Maintenance.MaintenancePublicConfigurationResource>> GetAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
 }
 namespace Azure.ResourceManager.Maintenance.Models
 {
+    public partial class MaintenanceConfigurationAssignment : Azure.ResourceManager.Models.ResourceData
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs]: fixed

File: sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs [hunk 3]
```diff
@@ -1,65 +1,29 @@
...
+        public virtual Azure.Response<Azure.ResourceManager.Maintenance.MaintenanceApplyUpdateResource> Get(string providerName, string resourceType, string resourceName, string applyUpdateName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.Maintenance.MaintenanceApplyUpdateResource>> GetAsync(string providerName, string resourceType, string resourceName, string applyUpdateName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
-    public partial class ApplyUpdateData : Azure.ResourceManager.Models.ResourceData
+    public partial class MaintenanceApplyUpdateData : Azure.ResourceManager.Models.ResourceData
     {
-        public ApplyUpdateData() { }
+        public MaintenanceApplyUpdateData() { }
         public System.DateTimeOffset? LastUpdateOn { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs]: fixed

File: sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yao725) [reply sdk/maintenance/Azure.ResourceManager.Maintenance/api/Azure.ResourceManager.Maintenance.netstandard2.0.cs:81]: ok, let's go header without changing these kinds of methods.

**Review by Yao725 — APPROVED** [review]

**Pull request merged**